samples/scanner/scanners/firewall_rules/blacklist.yaml

```diff
@@ -0,0 +1,154 @@
+# This file contains several example blacklist firewall rules.
+# 
+# Blacklist rules are evaluated in the following steps:
+# 1. Get list of all pairs of resources and policies that apply to the resource.
+# 2. For each pair, get a list of applicable rules by:
+#   a. Checking if a rule applies to the specific resource.
+#   b. Checking each parent in the hierarchy for a rule that applies to that parent.
+#     See: https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy
+#   c. Return the first applicable rules.
+# 3. For each applicable rule, check if the policy is a superset of one of the "match_policies" defined in the rule.
+# 4. For any rule with a matching policy, check if the policy is a superset of any of the "verify_policies"
+#     if it is, create a RuleViolation.
+rules:
+
+    # This rule will trigger on policies that allow ingress for any protocol or port.
+    #
+    # Example triggering firewall rule:
+    #   {
+    #     "name": "firewall1",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["0.0.0.0/0"],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "all",
+    #       }
+    #     ],
+    #     "direction": "INGRESS",
+    #   }
+    #
+    # Example firewall rule that WON'T trigger:
+    #   {
+    #     "name": "firewall2",
+    #     "network": "https://www.googleapis.com/compute/v1/projects/yourproject/global/networks/default",
+    #     "priority": 1000,
+    #     "sourceRanges": ["0.0.0.0/0"],
+    #     "allowed": [
+    #       {
+    #         "IPProtocol": "all",
+    #       }
+    #     ],
+    #     "direction": "EGRESS",
+    #   }
+  - rule_id: 'prevent_allow_all'
```

Comment:
nit: If you change the naming from the previous comment, please keep these examples in sync.